File: JavaScript/Codility/TapeEquilibrium.js
function solution(A) {
    var sumPartFirst = 0;
    var sumPartSecond = A.reduce((total, a) => total+a, 0);
    var answer = null;

    for (var i=0; i<A.length-1; i++){
        sumPartFirst += A[i];
        sumPartSecond -= A[i];
        
        var diffSum = Math.abs(sumPartFirst - sumPartSecond);
        if (answer === null || diffSum < answer){
            answer = diffSum;
        }
    }
    return answer;
}

File: Python/Codility/TapeEquilibrium.py
def solution(A):
    sumPartOne = 0
    sumPartTwo = sum(A)
    answer = None

    for p in range(len(A)-1):
        sumPartOne += A[p]
        sumPartTwo -= A[p]

        diffSum = abs(sumPartOne - sumPartTwo)
        answer = min(diffSum, answer) if answer is not None else diffSum

    return answer